code/aggregation/aggregate-raw-data.R

```diff
@@ -47,128 +47,127 @@ gm.afr.path <- file.path(data.dir, "GM", "gm-afr-inventories.tsv")
 }
 
 ## mark marginal phonemes in a boolean column and remove <angle brackets>
-markMarginal <- function(x) {
-    x$Marginal <- stri_detect_fixed(x$Phoneme, "<")
-    x$Phoneme <- stri_replace_all_fixed(x$Phoneme, replacement="", pattern="<")
-    x$Phoneme <- stri_replace_all_fixed(x$Phoneme, replacement="", pattern=">")
-    return(x)
+markMarginal <- function (df) {
+    df$Marginal <- stri_detect_fixed(df$Phoneme, "<")
+    df$Phoneme <- removeBrackets(df$Phoneme, "angle")
+    df
 }
 
-## propogate values through "long and sparse" data sources
-fillCells <- function(df, cols) {
-	for(col in cols) {
-		if(!is.na(match(col, colnames(df)))) {
-		    thiscol <- df[[col]]
-		    if(!is.na(thiscol[1])) thiscol <- na.locf(thiscol)
-		    df[[col]] <- thiscol
-		} else {
-			stop("Bad column name passed to 'fillCells'.")
-		}
-	}
-	return(df)
+## split character vectors and keep first element after split
+strsplitKeepFirst <- function (x, split.on) {
+    x <- sapply(stri_split_fixed(x, split.on), function (y) y[1])
 }
 
-## remove square brackets from allophones and collapse to a single cell
-collapseAllophones <- function(x, col) {
-	# x is the data frame, col is the name of the column to split on
-    # (col should typically be "InventoryID"). The first line splits
-    # the data by "col", then within "col" it splits again by SpecificDialect
-    # and then again by Phoneme.
-    spl <- split(x, x[[col]])
-    spt <- lapply(spl, function(i) {
-        if(length(unique(i$SpecificDialect)) > 1) {
-            dial <- split(i, i$SpecificDialect)
-            lect <- lapply(dial, function(j) {
-                phon <- split(j, j$Phoneme)
-                emes <- lapply(phon, function(k) {
-                    h <- k[1,]
-                    h$Allophones <- ifelse(is.na(h$Allophones), h$Phoneme,
-                                           paste(as.vector(k$Allophones),
-                                                 collapse=" "))
-                    h$Allophones <- stri_replace_all_fixed(h$Allophones,
-                                                           replacement="",
-                                                           pattern="[")
-                    h$Allophones <- stri_replace_all_fixed(h$Allophones,
-                                                           replacement="",
-                                                           pattern="]")
-                    tmp <- paste(as.vector(k$AllophoneNotes), collapse="; ")
-                    h$AllophoneNotes <- stri_replace_all_fixed(tmp,
-                                                               replacement="",
-                                                               pattern="\"")
-                    return(h)
-                    })
-                emes <- do.call(rbind, emes)
-                })
-            collapsed <- do.call(rbind, lect)
+## propogate values through "long and sparse" data sources
```

Comment:
the fillCells function (below) isn't new, just moved, and Git is doing a weird job of tracking that.